chainntnfs/txnotifier_test.go

```diff
@@ -1166,15 +1168,14 @@ func TestTxNotifierCancelConf(t *testing.T) {
 		TxIndex:     0,
 		Tx:          tx1,
 	}
-
-	// Before extending the notifier's tip with the block above, we'll
-	// cancel the second request.
-	n.CancelConf(ntfn2.HistoricalDispatch.ConfRequest, 2)
-
 	err = n.ConnectTip(block.Hash(), startingHeight+1, block.Transactions())
 	if err != nil {
 		t.Fatalf("unable to connect block: %v", err)
 	}
+
+	// Cancel the second notification before notifying to ensure its queued
+	// confirmation notification gets removed as well.
+	n.CancelConf(ntfn2.HistoricalDispatch.ConfRequest, 2)
```

Comment:
shouldn't we be testing both of these paths? cancel before `ConnectTip` and cancel after?

Reply:
The only difference between these paths is that an additional operation is done when canceling after, so we're already testing both paths by doing so.